packages/plugin-eslint/src/lib/meta/rules.spec.ts

```diff
@@ -0,0 +1,241 @@
+    cwdSpy = vi.spyOn(process, 'cwd');
+  });
+
+  afterAll(() => {
+    cwdSpy.mockRestore();
+  });
+
+  describe('React app', () => {
+    const appRootDir = join(fixturesDir, 'todos-app');
+    const eslintrc = join(appRootDir, '.eslintrc.js');
+
+    const eslint = new ESLint({
+      useEslintrc: false,
+      baseConfig: { extends: eslintrc },
+    });
+    const patterns = ['src/**/*.js', 'src/**/*.jsx'];
+
+    beforeAll(() => {
+      cwdSpy.mockReturnValue(appRootDir);
+    });
+
+    it('should list expected number of rules', async () => {
+      await expect(listRules(eslint, patterns)).resolves.toHaveLength(47);
+    });
+
+    it('should include explicitly set built-in rule', async () => {
+      await expect(listRules(eslint, patterns)).resolves.toContainEqual({
+        ruleId: 'no-const-assign',
+        meta: {
+          docs: {
+            description: 'Disallow reassigning `const` variables',
+            recommended: true,
+            url: 'https://eslint.org/docs/latest/rules/no-const-assign',
+          },
+          messages: {
+            const: "'{{name}}' is constant.",
+          },
+          schema: [],
+          type: 'problem',
+        },
+        options: [],
+      } satisfies RuleData);
+    });
+
+    it('should include explicitly set plugin rule', async () => {
+      await expect(listRules(eslint, patterns)).resolves.toContainEqual({
+        ruleId: 'react/jsx-key',
+        meta: {
+          docs: {
+            category: 'Possible Errors',
+            description:
+              'Disallow missing `key` props in iterators/collection literals',
+            recommended: true,
+            url: 'https://github.com/jsx-eslint/eslint-plugin-react/tree/master/docs/rules/jsx-key.md',
+          },
+          messages: expect.any(Object),
+          schema: [
+            {
+              additionalProperties: false,
+              properties: {
+                checkFragmentShorthand: { default: false, type: 'boolean' },
+                checkKeyMustBeforeSpread: { default: false, type: 'boolean' },
+                warnOnDuplicates: { default: false, type: 'boolean' },
+              },
+              type: 'object',
+            },
+          ],
+        },
+        options: [],
+      } satisfies RuleData);
+    });
+  });
+
+  describe('Nx monorepo', () => {
+    const nxRootDir = join(fixturesDir, 'nx-monorepo');
+    const eslintrc = join(nxRootDir, 'packages/utils/.eslintrc.json');
+
+    const eslint = new ESLint({
+      useEslintrc: false,
+      baseConfig: { extends: eslintrc },
+    });
+    const patterns = ['packages/utils/**/*.ts', 'packages/utils/**/*.json'];
+
+    beforeAll(() => {
+      cwdSpy.mockReturnValue(nxRootDir);
+    });
+
+    it('should list expected number of rules', async () => {
+      await expect(listRules(eslint, patterns)).resolves.toHaveLength(69);
+    });
+
+    it('should include explicitly set plugin rule with custom options', async () => {
```

Comment:
It could be valuable to add a check for a case where the same rule but with different custom options targetting different file pattern is present in ESLint config.

Reply:
I'll revisit this later. I tried, but it's pretty complex and I don't want to prolong this PR too much. There will be more ESLint-related PRs.